src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -187,80 +197,96 @@ protected override bool Receive(object message)
-                
+
                 case KafkaConsumerActorMetadata.Internal.ISubscriptionRequest subscribe:
                     HandleSubscription(subscribe);
                     return true;
-                
+
+                case KafkaConsumerActorMetadata.Internal.RegisterSubStage subStage:
+                    _stageActorsMap = _stageActorsMap.SetItem(subStage.TopicPartitions, Sender);
+                    return true;
+
                 case KafkaConsumerActorMetadata.Internal.RequestMessages requestMessages:
-                    if(_settings.VerboseLogging)
-                        _log.Debug("Messages was requested, RequestId: {0}, Partitions: {1}", requestMessages.RequestId, string.Join(", ", requestMessages.Topics));
+                    if (_settings.VerboseLogging)
+                        _log.Debug("Messages was requested, RequestId: {0}, Partitions: {1}", requestMessages.RequestId,
+                            string.Join(", ", requestMessages.Topics));
                     Context.Watch(Sender);
                     CheckOverlappingRequests("RequestMessages", Sender, requestMessages.Topics);
-                    _requests = _requests.SetItem(Sender, requestMessages);
-                    _requestors = _requestors.Add(Sender);
-                    
+
+                    if (_stageActorsMap.GetOrElse(requestMessages.Topics, Sender).Equals(Sender))
+                        _requests = _requests.SetItem(Sender, requestMessages);
+
                     // When many requestors, e.g. many partitions with committablePartitionedSource the
                     // performance is much by collecting more requests/commits before performing the poll.
                     // That is done by sending a message to self, and thereby collect pending messages in mailbox.
-                    if (_requestors.Count == 1)
+                    if (_stageActorsMap.Count == 1)
                     {
                         Poll();
                     }
-                    else if (!_delayedPollInFlight)
+                    else
                     {
-                        _delayedPollInFlight = true;
-                        Self.Tell(_delayedPollMessage);
+                        RequestDelayedPoll();
                     }
+
                     return true;
-                
+
                 case KafkaConsumerActorMetadata.Internal.Seek seek:
-                    foreach (var offset in seek.Offsets)
+                    try
                     {
-                        _seekedOffset = _seekedOffset.SetItem(offset.TopicPartition, offset);
+                        foreach (var offset in seek.Offsets)
+                        {
+                            _seekedOffset = _seekedOffset.SetItem(offset.TopicPartition, offset);
+                        }
+
+                        Sender.Tell(Done.Instance);
+                    }
+                    catch (Exception ex)
+                    {
+                        SendFailure(ex, Sender);
```

Comment:
Important detail: if we fail to `Seek` then we send that back as a `Status.Failure` to the `StageActor` now - this will cause that stage to fail and terminate.